measurements/pages/__init__.py

```diff
@@ -234,14 +235,22 @@ def generator():
 
                     # Sanity checks to ensure the streamed data start with
                     # `{` and ends with `\n`
-                    if beginning:
-                        assert d[:1] == b'{', 'Chunk starts with %r != {' % d[:1]
-                    if to_read == len(d):
-                        assert d[-1:] == b'\n', 'Chunk ends with %r != \\n' % d[-1:]
+                    if beginning and d[:1] != b'{':
+                        raise HTTPException('Chunk starts with %r != {' % d[:1])
+                    if to_read == len(d) and d[-1:] != b'\n':
+                        raise HTTPException('Chunk ends with %r != \\n' % d[-1:])
 
                     yield d
                     to_read -= len(d)
-                    beginning = False
+                    if len(d): # valid lz4 frame may have 0 bytes
+                        beginning = False
+            # `autoclaved` file format may have `\n` in separate LZ4 frame,
+            # database stores offset for JSON blobs without trailing newline,
+            # here is hack adding newline as next frame boundaries are unknown.
+            if r.raw.read(1) != b'': # stream must be already EOFed
```

Comment:
I thinks this should actually never happen, but I guess it's ok to check just in case (note: I was initially confused by this and thought it was somehow related to the change of the while clause being `1` instead of `0`)

Reply:
It catched off-by-one error in Range :)